services/posting.js

```diff
@@ -21,7 +21,20 @@ const createPosting = async (req) => {
     return createdPost
 }
 
+const getPosting = async (req, res) => {
+    const {
+        params: { id: username }
+    } = req
+
+    const posting = await Posting.findOne({ _id: username })
```

Comment:
Two fixes here. First we want to get all posts for a user not just one. 
Second we search be the username, and the id is not the username. Mongoose is smart so we can just
```
    const posting = await Posting.find({ username })
```